### PR TITLE
LIME-437 Template conversion realignment and update

### PR DESCRIPTION
**Pull request by smsgds**
### What changed

The PR aligns branch LIME-137 with main branch. see https://github.com/alphagov/di-ipv-cri-uk-passport-back/compare/LIME-131-TemplateConversion...LIME-437

- Revert splitting configuration service parameters into environment and stack - all now stackbased.
- remove EnvironmentVariable ENVIRONMENT (no longer used) - Note Runtime ENV VAR
- Log group naming determined by lambda function
- TLS related paramters moved under /DCS/HTTPClient
- DCS JWS parameters moved under /DCS/JWS
- DCS JWE parameters moved under /DCS/JWE
- getClientIssuer, getClientRedirectUrls and getClientSigningPublicJwk changed to use AWS_STACK_NAME directly instead of CREDENTIAL_ISSUERS_CONFIG_PARAM_PREFIX
- Removed CREDENTIAL_ISSUERS_CONFIG_PARAM_PREFIX as it was set to AWS_STACK_NAME (which was originally ENVIRONMENT)
- Align deploy.sh helper script, removed PackageType: Zip and update template codeURI location with common-cri dev deployment method
- note common CRI framework is not used in passport
- Updated converted template to enable metrics and alarms.
- Add PermissionsBoundary missing from template Parameters section.
- No longer deletes original deploy/template.yaml
- update readme for deployments

### Why did it change

Originally LIME-137 was deployed into an existing  environment. The existing stack located parameters using environment name. Reuse of these parameters was needed to deploy a new stack, while the new template based parameters where stackbased. This change has been reverted so that all parameters can no be located via stackname. Stack based parameters avoid clashes in dev with multiple stacks.

Log group naming compatibility is no longer needed due to being deployed to a new environment and aligns with the other CRI's.

HTTP/JWS/JWE parameters relocated to help avoid confusion as to which certs relate to which part of the DCS connection. See ConfigurationVariable.java

deploy.sh update to ease later transition to using common-lambda stack.

PermissionsBoundary was missed in LIME-137 but found to be needed post merge in build env.

Post LIME-137 ported to new template.
Metrics and Alarms added 
Subscription Filter destinations
JwtTtlUnitParameter/MaxVCJwtTtlMapping

### Issue tracking

- [LIME-437](https://govukverify.atlassian.net/browse/LIME-437)

[LIME-437]: https://govukverify.atlassian.net/browse/LIME-437?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: infrastructure/lambda/template.yaml [hunk 1]
```diff
@@ -0,0 +1,1384 @@
...
+    Environment:
+      dev: 512
+      build: 1024
+      staging: 1024
+      integration: 1024
+      production: 2048
+
+  SessionTtlMapping:
+    Environment:
+      dev: "172800" # 2 days
```
Comment (gds-jbentham) [inline infrastructure/lambda/template.yaml:121]: How come this is 2 days rather than 2 hours?

File: infrastructure/lambda/template.yaml [hunk 2]
```diff
@@ -0,0 +1,1384 @@
...
+      integration: "ewogICAgImt0eSI6ICJFQyIsCiAgICAidXNlIjogInNpZyIsCiAgICAiY3J2IjogIlAtMjU2IiwKICAgICJraWQiOiAiaXB2LWNvcmUtc3R1Yi0yLWZyb20tbWtqd2sub3JnIiwKICAgICJ4IjogImszOXVLYWNTdWtRQnJNWnJIRFRCVVpzbGl2cFhLRE5aVGc2aW5DSHdyTGMiLAogICAgInkiOiAiOEY4TG5RN3dHOWh4c1Q0YXgwQXR5N2lNR0l5aVlfWUdwM19xSVp6S28xQSIsCiAgICAiYWxnIjogIkVTMjU2Igp9"
+
+  IPVCore1PublicSigningJwkBase64Mapping:
+    Environment:
+      staging: "eyJrdHkiOiJFQyIsImNydiI6IlAtMjU2IiwieCI6ImtlMVRNRnFNb0Z5eHg1eXpOdFFRbGw0dk9yeHZUdFBKQ0huUzRqOHpoMlUiLCJ5IjoicURLX0g4QXpKS2FIbU1zaHg5TGp2LTB0ek5rV2EtSkVHUzJtZHRKUjFPQSJ9"
+      integration: "eyJrdHkiOiJFQyIsImNydiI6IlAtMjU2IiwieCI6IkJUUWdWQjU0RE9JcDU0eGRVSVg0SGtUX3pCdjZHdVdMV1RUTkdxMk15dEkiLCJ5IjoiTFFRamx5ZEtOMUhXZFJQcFBpalJObEJrbi1qaDgzZzBBUmIyNms2WVh1byJ9"
+      production: "eyJrdHkiOiJFQyIsImNydiI6IlAtMjU2IiwieCI6IlVQdlU1TlBtRUxyV2lXU01WZkREN0c4dTNFSllyeXFQSVo0Nlc5TUFsUmMiLCJ5Ijoicjc3RjItS1BocHZUSUdFV2d0NVNtYXZTdkJVSENxV1V4RDZSR19GSkhWayJ9"
+
+  IPVCoreStubRedirectURIMapping:
```
Comment (gds-jbentham) [inline infrastructure/lambda/template.yaml:204]: In other CRIs for dev we use       
dev: "https://di-ipv-core-stub.london.cloudapps.digital/callback"

File: infrastructure/lambda/template.yaml [hunk 3]
```diff
@@ -0,0 +1,1384 @@
...
+    # checkov:skip=CKV_AWS_26: We will update this once basic alerting is available
+    Metadata:
+      SamResourceId: AlarmTopicPassport
+  AlarmTopicSubscriptionPagerDutyPassport:
+    Type: AWS::SNS::Subscription
+    Properties:
+      TopicArn:
+        Ref: AlarmTopicPassport
+      Endpoint:
+        Fn::Sub: '{{resolve:ssm:/alerting/pagerduty-passport/url}}'
```
Comment (gds-jbentham) [inline infrastructure/lambda/template.yaml:1322]: This param will need set in envs prior to deploy can we get a task in place for this. We also may need another pagerduty service can we raise another task to discuss

File: lib/src/main/java/uk/gov/di/ipv/cri/passport/library/config/ConfigurationVariable.java [hunk 4]
```diff
@@ -1,31 +1,29 @@
...
-    PASSPORT_CRI_SIGNING_CERT("/%s/credentialIssuers/ukPassport/self/signingCertForDcsToVerify"),
-    PASSPORT_CRI_SIGNING_KEY("/%s/credentialIssuers/ukPassport/self/signingKeyForPassportToSign"),
-    PASSPORT_CRI_TLS_CERT("/%s/credentialIssuers/ukPassport/self/tlsCert"),
-    PASSPORT_CRI_TLS_KEY("/%s/credentialIssuers/ukPassport/self/tlsKey"),
-    VERIFIABLE_CREDENTIAL_ISSUER(
-            "/%s/credentialIssuers/ukPassport/self/verifiableCredentialIssuer"),
-    VERIFIABLE_CREDENTIAL_SIGNING_KEY_ID(
-            "/%s/credentialIssuers/ukPassport/self/verifiableCredentialKmsSigningKeyId");
+            "/%s/DCS/JWE/EncryptionKeyForPassportToDecrypt"), // DCS JWE (Private Key Reply Decrypt)
+    JAR_ENCRYPTION_KEY_ID("/%s/AuthRequestKmsEncryptionKeyId"),
```
Comment (gds-jbentham) [inline lib/src/main/java/uk/gov/di/ipv/cri/passport/library/config/ConfigurationVariable.java:18]: Can we check is there a convention should params be named like this like classes or should they start small and be camel cased

Comment (gds-jbentham) [reply lib/src/main/java/uk/gov/di/ipv/cri/passport/library/config/ConfigurationVariable.java:18]: Also should align with any common naming to make the swap easier where we can

File: infrastructure/lambda/template.yaml (hunk 1 above)
Comment (smsgds) [reply infrastructure/lambda/template.yaml:121]: This part of the template is aligned pre-common-cri-framework to 2 days.

Originally in passport this was 1 day (named backendSessionTtl), and the other CRI's had it at the time set to 2 days as SessionTtl

```
  backendSessionTtl:
    Type: AWS::SSM::Parameter
    Properties:
      Name: !Sub /${Environment}/credentialIssuers/ukPassport/self/backendSessionTtl
      Type: String
      Value: 86400
```

FraudCRI removing the SessionTtlMapping from template when it was 2 days.
https://github.com/alphagov/di-ipv-cri-fraud-api/commit/cc99b257eb32b570a678743634043892fa52fcb3

Checking the common-cri template SessionTtl is now set to 2 hours, I was unaware of this reduction.

File: infrastructure/lambda/template.yaml (hunk 2 above)
Comment (smsgds) [reply infrastructure/lambda/template.yaml:204]: Passport/Jar validator checks the redirect_uri and client in the session request against this expected one. I would need to investigate how the mismatch does not occur in the other CRI's.

The validation error is
Invalid redirect_uri claim (http://.....) provided for client: .....

File: lib/src/main/java/uk/gov/di/ipv/cri/passport/library/config/ConfigurationVariable.java (hunk 4 above)
Comment (smsgds) [reply lib/src/main/java/uk/gov/di/ipv/cri/passport/library/config/ConfigurationVariable.java:18]: There is heavy inconsistency in template variable between all the CRI's and in the common-framework.

The template variables are named consistently with this inconsistency if you will, so when swapped they should all be read from the same location.

In common-lambda template, variables are majority Title Case, just after the stack name.
jwtAuthentication related paths are lower case with variables in camel case aligned with other all other CRI's and common-lambda.

The passport specific variables are Title Case, just after the stack name most will be replaced with common-lambdas and match up with them common lambdas locations.

DCS params are Title Case but sorted out into paths with 3 letter acronyms to help different the different keys and certs from each other.

File: infrastructure/lambda/template.yaml (hunk 1 above)
Comment (smsgds) [reply infrastructure/lambda/template.yaml]: The SessionTtlMapping values are reduced to 7200.
A small investigation into the TTL found that the backendSessionTtl did operate in the same way as SessionTtl in common.
A review of TTL's has taken place with the VC ttl values now the same as other CRI's and the passport specific AccessTokenReqJwtTtl exposed.

File: infrastructure/lambda/template.yaml (hunk 2 above)
Comment (gds-jbentham) [reply infrastructure/lambda/template.yaml:205]: It will be a parameter issue lets pickup once deployed

**Review by gds-jbentham — APPROVED** [review]

**Pull request merged**